### ISSUE TITLE
node-csv-parse is 5x slower than other parsers

### ISSUE THREAD

**Issue by phihag** [wdavidw/node-csv-parse#106]
At least for me, node-csv-parse is significantly slower than other comparable parsers. Is there a way to speed it up, or am I misusing it? Are there any options to improve performance? Here's my code:

```
var csv_parse = require('csv-parse');
var fs = require('fs');

var read_stream = fs.createReadStream('data.csv', {encoding: 'utf8'});
var parser = csv_parse({}, function(err, lines) {
    if (err) throw err;
    console.log(lines.length + ' lines');
});
read_stream.pipe(parser);
```

You can see the whole source code of the test (including test data generation, just run `make`) at https://github.com/phihag/csv-speedtest . On the test data set on my machine, csv-parse clocks in at about 1.3s when I use the string interface, and 1.4s with streams. For comparison, papaparse takes ~0.25s, and Python's csv ~0.3s.

**Comment by wdavidw** [comment]
There are probably a few ways to improve performance. I havent spend much time investigating it but we shall start looking at how v8 optimize the all process to ensure we leverage the engine.

**Comment by x3cion** [comment]
Hi,

1.1.3 does not have my performance regression fix pulled in with #105 . Maybe it's time for a new release?

I've seen other deoptimization reasons by V8, but it's hard to find them, because the functions are long and not split up.

(I'm waiting for a new release too. :) )

**Comment by wdavidw** [comment]
closing, feel free to re-open if necessary
